synchronizer/synchronizer.go

```diff
@@ -315,6 +333,20 @@ func (s *Synchronizer) AdvanceView(syncInfo hotstuff.SyncInfo) {
 		}
 	}
 
+	b, ok := syncInfo.Block()
+	if ok {
+		s.updateLeafBlock(b)
+	}
+
+	if s.isInPipelineStretch(s.leafBlock.View()) {
+		v = s.leafBlock.View()
+	}
+
+	if timeoutV, ok := syncInfo.TimeoutView(); ok &&
```

Comment:
Returning  `ok` is not necessary here. You can just do:
```go
	if timeoutV := syncInfo.TimeoutView(); timeoutV > v && s.isInPipelineStretch(timeoutV) {
```
Since `timeoutV == 0` will render the if-condition false, as with the current `ok` version.

Reply:
fixed in [aae49db](https://github.com/relab/hotstuff/pull/78/commits/aae49db4ba31404475852512adf01459e04e32f4)
```go
// TimeoutView returns the timeout view, if present.
func (si SyncInfo) TimeoutView() (_ View) {
	return si.timeoutView
}
```